### PR TITLE
Vp 1186 validation for required fields

### PR DESCRIPTION
**Pull request by AliveMen**
fix form validation

### CODE REVIEW

File: VirtoCommerce.JavaScriptShoppingCart.Web/Scripts/checkout/checkout-address.js [hunk 1]
```diff
@@ -58,14 +58,15 @@ cartModule.component('vcCheckoutAddress', {
 				ctrl.address.regionName = undefined;
 			}
 		}
+		ctrl.setForm = function (frm) { ctrl.form = frm; };
 
 		ctrl.validate = function () {
```
Comment (t13ka) [inline VirtoCommerce.JavaScriptShoppingCart.Web/Scripts/checkout/checkout-address.js:63]: Is this ok that func Validate returns a value? Usually, validate function shouldn't return any value, just void

File: VirtoCommerce.JavaScriptShoppingCart.Web/Scripts/checkout/checkout-address.js [hunk 2]
```diff
@@ -58,14 +58,15 @@ cartModule.component('vcCheckoutAddress', {
...
 			if (ctrl.form) {
 				ctrl.form.$setSubmitted();
 				return !ctrl.form.$invalid;
 			}
 			return true;
-		}
+		};
 
 		function stringifyAddress(address) {
 			var stringifiedAddress = address.firstName + ' ' + address.lastName + ', ';
```
Comment (t13ka) [inline VirtoCommerce.JavaScriptShoppingCart.Web/Scripts/checkout/checkout-address.js]: I guess the code will be shorter in this context if use 'result' instead 'stringifiedAddress'. It's valid cuz this func doing only formating and nothing more... We know from the name of the function that the result is stringified and there are no reasons mark our result with this adjective

Comment (AliveMen) [reply VirtoCommerce.JavaScriptShoppingCart.Web/Scripts/checkout/checkout-address.js]: fixed

File: VirtoCommerce.JavaScriptShoppingCart.Web/Scripts/checkout/checkout-address.js (hunk 1 above)
Comment (AliveMen) [reply VirtoCommerce.JavaScriptShoppingCart.Web/Scripts/checkout/checkout-address.js:63]: I agree with you. Possible is a better way to call it IsValid. But here a lot of changes in different checkout-* blades. https://github.com/VirtoCommerce/vc-module-javascript-shoppingcart/blob/a883fa31be61d483613386f84150f2e8958ff97a/VirtoCommerce.JavaScriptShoppingCart.Web/Scripts/checkout/checkout-wizard-step.js#L32. 
It's out of the scope of this fix.

**Review by t13ka — APPROVED** [review]

**Pull request merged**